includes/resources/rdsm_conversion.php

```diff
@@ -63,13 +63,42 @@ public function build_payload($form_data) {
 
   private function filter_fields(array $ignored_fields, $form_fields){
     foreach ($form_fields as $field => $value) {
-      if(in_array($field, $ignored_fields)){
+      if (in_array($field, $ignored_fields)) {
+        unset($form_fields[$field]);
+      }
+
+      if ($this->is_credit_card_number($value)) {
         unset($form_fields[$field]);
       }
     }
+
     return $form_fields;
   }
 
+  private function is_credit_card_number($number)  {
+    $number = preg_replace('/\D/', '', $number);
+
+    $cardtype = array(
+      "amex"       => "/^3[47][0-9]{13}$/",
+      "diners"     => "/^3(?:0[0-5]|[68][0-9])[0-9]{11}$/",
+      "discover"   => "/^6(?:011|5[0-9]{2})[0-9]{12}$/",
+      "elo"        => "/^((((636368)|(438935)|(504175)|(451416)|(636297)(506699)|(636369))\d{0,10})|((5067)|(4576)|(4011))\d{0,12})/",
+      "mastercard" => "/^5[1-5][0-9]{14}$/",
+      "visa"       => "/^(5[1-5]|222[1-9]|22[3-9]|2[3-6]|27[01]|2720)[0-9]{14}$/"
+    );
+
+    if (preg_match($cardtype['amex'], $number) ||
+        preg_match($cardtype['diners'], $number) ||
+        preg_match($cardtype['discover'], $number) ||
+        preg_match($cardtype['elo'], $number) ||
+        preg_match($cardtype['mastercard'], $number) ||
+        preg_match($cardtype['visa'], $number)) {
+      return true;
+    } else {
+      return false;
+    }
```

Comment:
What do you think about looping all the card types to validate them?

This way, when we add a new card type, we'll only need to change the `$cardtypes` array.
```suggestion
foreach($cardtypes as $cardtype) {
    $credit_card_number = preg_match($cardtype, $number);

    if ($credit_card_number) {
      return true;
    }
  }
  
  return false;
```